nginx.tmpl

```diff
@@ -379,3 +379,142 @@ server {
 
 {{ end }}
 {{ end }}
+
+# VIRTUAL_HOST_ALIAS
+{{ range $host_alias, $containers := groupByMulti $ "Env.VIRTUAL_HOST_ALIAS" "," }}
+
+{{ $host_alias := trim $host_alias }}
+
+{{ $first_host := (first (groupByKeys $containers "Env.VIRTUAL_HOST")) }}
+{{ $first_host := trim $first_host }}
+
+# First Host {{ $first_host }}
+
+#Alias:  {{ $host_alias }}
+
+{{ $default_host := or ($.Env.DEFAULT_HOST) "" }}
+{{ $default_server := index (dict $host_alias "" $default_host "default_server") $host_alias }}
+
+{{/* Get the NETWORK_ACCESS defined by containers w/ the same vhost, falling back to "external" */}}
+{{ $network_tag := or (first (groupByKeys $containers "Env.NETWORK_ACCESS")) "external" }}
+
+{{/* Get the HTTPS_METHOD defined by containers w/ the same vhost, falling back to "redirect" */}}
+{{ $https_method := or (first (groupByKeys $containers "Env.HTTPS_METHOD")) "redirect" }}
+
+{{/* Get the SSL_POLICY defined by containers w/ the same vhost, falling back to empty string (use default) */}}
+{{ $ssl_policy := or (first (groupByKeys $containers "Env.SSL_POLICY")) "" }}
+
+{{/* Get the HSTS defined by containers w/ the same vhost, falling back to "max-age=31536000" */}}
+{{ $hsts := or (first (groupByKeys $containers "Env.HSTS")) "max-age=31536000" }}
+
+{{/* Get the first cert name defined by containers w/ the same vhost */}}
+{{ $certName := (first (groupByKeys $containers "Env.CERT_NAME")) }}
+
+{{/* Get the best matching cert  by name for the vhost. */}}
+{{ $vhostCert := (closest (dir "/etc/nginx/certs") (printf "%s.crt" $host_alias))}}
+
+{{/* vhostCert is actually a filename so remove any suffixes since they are added later */}}
+{{ $vhostCert := trimSuffix ".crt" $vhostCert }}
+{{ $vhostCert := trimSuffix ".key" $vhostCert }}
+
+{{/* Use the cert specified on the container or fallback to the best vhost match */}}
+{{ $cert := (coalesce $certName $vhostCert) }}
+
+{{ $is_https := (and (ne $https_method "nohttps") (ne $cert "") (exists (printf "/etc/nginx/certs/%s.crt" $cert)) (exists (printf "/etc/nginx/certs/%s.key" $cert))) }}
+
+{{ if $is_https }}
+
+{{ if eq $https_method "redirect" }}
+server {
+	server_name {{ $host_alias }};
+	listen 80 {{ $default_server }};
+	{{ if $enable_ipv6 }}
+	listen [::]:80 {{ $default_server }};
+	{{ end }}
+	access_log /var/log/nginx/access.log vhost;
+	return 301 https://{{ $first_host }}$request_uri;
+}
```

Comment:
I think the http and the https server blocks can be merged into a single one.
Besides that it would be good to support the new variable http and https port variable. (rebase to current master to get these)
The new block could look something like that:
```
server {
￼	server_name {{ $host_alias }};
        {{ if eq $https_method "redirect" }}
        listen {{ $external_http_port }} {{ $default_server }};
        {{ end }}
￼	listen {{ $external_https_port }} ssl http2 {{ $default_server }};
￼	{{ if $enable_ipv6 }}
        {{ if eq $https_method "redirect" }}
        listen [::]:{{ $external_http_port }} {{ $default_server }};
        {{ end }}
￼	listen [::]:{{ $external_https_port }} ssl http2 {{ $default_server }};
￼	{{ end }}
￼	access_log /var/log/nginx/access.log vhost;
# ... (rest of the ssl block)
```

Reply:
Thanks for the review @aiomaster . I've merged the 2 blocks together.

Do you also suggest replacing all other instances where http and https hardcoded ports with the new variables?

Reply:
Yes the ports 80 and 443 should be replaced with the corresponding variables everywhere. My snippet was just an example and not a complete full review. Thanks for updating this.